### PR TITLE
added location and locations printouts to remote and SMPC-encrypted nn.Module

### PR DESCRIPTION
**Pull request by NiWaRe**
## Description
This adds a new `__str__()` function to `torch.nn.Module` which adds a location printout to every shared model. In the case of an SMPC-encrypted model it outputs all the different locations. Should serve as a small convenience function.

## Affected Dependencies
None.

## How has this been tested?
- Tested the output of normal tensors, a remote tensor, a SMPC-encrypted tensor, a normal nn.Module, a nn.Module with FixedPrecisionTensor weights, a remote nn.Module and an SMPC-encrypted nn.Module.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by NiWaRe** [comment]
There are these three other issues that also exist, I could take a look or, given that I already have another task, leave it for now. What do you think @LaRiffle @gmuraru? 

* Further possible tasks 
  1. add 'locations' property to additiveSharedModels (same as 'location' for normal remote models)
  2. suppress 'location' property in normal tensors 
  3. FixedPrecisionTensor.child has a location property (should only have if AdditiveSharedTensor as child)

**Review by LaRiffle — APPROVED** [review]
Perfect! Thanks for the fix :)
